src/main/java/com/nha/abdm/wrapper/common/dataPackaging/encryption/EncryptionController.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.asn1.x9.X9ECParameters;
+import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.crypto.ec.CustomNamedCurves;
+import org.bouncycastle.crypto.engines.AESEngine;
+import org.bouncycastle.crypto.generators.HKDFBytesGenerator;
+import org.bouncycastle.crypto.modes.GCMBlockCipher;
+import org.bouncycastle.crypto.params.AEADParameters;
+import org.bouncycastle.crypto.params.HKDFParameters;
+import org.bouncycastle.crypto.params.KeyParameter;
+import org.bouncycastle.jce.interfaces.ECPublicKey;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jce.spec.ECParameterSpec;
+import org.bouncycastle.jce.spec.ECPrivateKeySpec;
+import org.bouncycastle.jce.spec.ECPublicKeySpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@Service
+public class EncryptionController {
+  private static final Logger log = LogManager.getLogger(EncryptionController.class);
+  @Autowired KeyController keyController;
+
+  public EncryptionResponse encrypt(
+      @RequestBody HIPHealthInformationRequest hipHealthInformationRequest,
+      BundleResponseHIP bundleResponse)
+      throws Exception {
+    log.info(bundleResponse.getBundleContent());
+    log.info("<---------------- Encryption started ------------------->");
+    KeyMaterial senderKeys = keyController.fetchKeys();
+    KeyMaterial receiverKeys =
+        KeyMaterial.builder()
+            .publicKey(
+                hipHealthInformationRequest
+                    .getHiRequest()
+                    .getKeyMaterial()
+                    .getDhPublicKey()
+                    .getKeyValue())
+            .nonce(hipHealthInformationRequest.getHiRequest().getKeyMaterial().getNonce())
+            .build();
+    byte[] xorOfRandom = xorOfRandom(senderKeys.getNonce(), receiverKeys.getNonce());
+
+    String encryptedData =
+        encrypt(
+            xorOfRandom,
+            senderKeys.getPrivateKey(),
+            receiverKeys.getPublicKey(),
+            bundleResponse.getBundleContent());
+
+    String keyToShare = getBase64String(getEncodedHIPPublicKey(getKey(senderKeys.getPublicKey())));
+    return new EncryptionResponse(encryptedData, keyToShare, senderKeys.getNonce());
+  }
+
+  private byte[] xorOfRandom(String senderNonce, String receiverNonce) {
+    byte[] randomSender = getBytesForBase64String(senderNonce);
+    byte[] randomReceiver = getBytesForBase64String(receiverNonce);
+
+    byte[] combinedRandom = new byte[randomSender.length];
+    for (int i = 0; i < randomSender.length; i++) {
+      combinedRandom[i] = (byte) (randomSender[i] ^ randomReceiver[i % randomReceiver.length]);
+    }
+    return combinedRandom;
+  }
+
+  public byte[] getBytesForBase64String(String value) {
```

Comment:
lets restrict the access modifier to private unless its required outside package